Spec/Tests/AuthTests.swift

```diff
@@ -2970,18 +2936,11 @@ class AuthTests: XCTestCase {
         }
     }
 
-    func test__107__authorize__should_adhere_to_all_requirements_relating_to__authUrl_with_json() {
-        guard let tokenDetails = getTestTokenDetails() else {
-            XCTFail("TokenDetails is empty")
-            return
-        }
-
-        let encoder = ARTJsonLikeEncoder()
```

Comment:
I was wondering how this change (extracting the `encoder` variable) is related to the rest of the changes in this commit (using `XCTUnwrap`)? Also, could you explain the motivation for the change? It introduces shared state between test cases, which is something we should be trying to _reduce_, not add, I think.

Reply:
About motivation behind this change - https://github.com/ably/ably-cocoa/pull/1258#discussion_r775505729
Since `ARTJsonLikeEncoder`/`ARTJsonEncoder` doesn't keep any state by themselves and are merely just a wrappers around `NSJSONSerialization` and `NSDictionary` I think it's fine to hide them behind this global var. WDYT @lukasz-szyszkowski  @QuintinWillison

Reply:
[Here](https://github.com/ably/ably-cocoa/pull/1258#discussion_r780922925) is my comment

Reply:
I think it still adds mental overhead, somebody has to know that those classes don't have any internal state, and it's an undocumented assumption which we have to make sure we don't violate by accident. I don't think it costs much to just create a new one each time:

```swift
private func createJsonEncoder() -> ARTJsonLikeEncoder {
     let encoder = ARTJsonLikeEncoder()
     encoder.delegate = ARTJsonEncoder()
     return encoder
 }
```

and in tests

```swift
let jsonTokenDetails = try XCTUnwrap(createJsonEncoder().encode(testTokenDetails), "Invalid TokenDetails")
```

I think that if we didn't already have loads of shared state in the tests, it would bother me less, but since doing #1240 we have a lot already.